vectorMath.py

```diff
@@ -0,0 +1,79 @@
+import math
+
+
+def magnitude(u):
+    index = 0
+    sum = 0
+    for i in u:
+        mag = (u[index])**2
+        index += 1
+        sum = sum + mag
+    return math.sqrt(sum)
+
+
+def dotProduct(u, v):
+    index = 0
+    sum = 0
+    for i in u:
+        dot = (u[index] * v[index])
+        sum = sum + dot
+        index += 1
+    return sum
+
+
+def angleBtwnVectors(u, v):
+    theta = math.acos((dotProduct(u, v))/((magnitude(u))*(magnitude(v))))
+    return math.degrees(theta)
+
+
+def crossProduct(u, v):
+    cross = [(u[1]*v[2]-v[1]*u[2]),
+             (v[0]*u[2]-u[0]*v[2]),
+             (u[0]*v[1]-u[1]*v[0])]
+    return cross
+
+
+choice = raw_input('You can choose:\n'
+                   'a) Magnitude\n'
+                   'b) Angle btwn two vectors\n'
+                   'c) Dot Product\n'
+                   'd) Cross Product\n')
+
+if choice == 'a':
+    print 'You chose magnitude!'
+    u = raw_input("Enter vector u seperated by spaces: ")
+    u = map(int, u.split())
+    print "Magnitude of vector u = ", magnitude(u)
```

Comment:
The more canonical way for doing this is to use [.format()](https://docs.python.org/3.4/library/string.html#format-string-syntax) on a string, in this case this whole line would be:
```
    print("Magnitude of vector u = {}".format(magnitude(u)))
```

Reply:
`.format()` is actually falling out of favor in python 3 as well. The best way now is to put a leading f in front of the string. Then, anything in curly braces will have access to any variable in the scope and believe it or not even execute any valid python code. For example:
```
x = 'foo'
y = 40
z = 'bar'
print(f"I have {y + 2} {x}s but a {z} ain't one")
```